src/GNDStk/Node/src/meta.hpp

```diff
@@ -0,0 +1,132 @@
+
+// -----------------------------------------------------------------------------
+// Node::meta(string)
+//
+// Searches the Node's metadata for a metadatum of the given name. If found,
+// returns the key's paired value. Else, returns an empty string.
+// -----------------------------------------------------------------------------
+
+// const
+const std::string &meta(
+   const std::string &key,
+   bool &found = detail::default_bool
+) const {
+   // search
+   for (auto &m : metadata)
+      if (m.first == key)
+         return found = true, m.second;
```

Comment:
A consideration to keep in mind (which I do in the system I'm building) is that many particular child nodes have to be unique as well. Not because the file format (e.g. XML) requires it (XML doesn't), but because GNDS may require it.